lib/commands/shell.js

```diff
@@ -9,19 +10,38 @@ commands.mobileShell = async function (opts = {}) {
     log.errorAndThrow(`Appium server must have relaxed security flag set in order to run any shell commands`);
   }
 
-  if (!_.isString(opts.command)) {
+  const {
+    command,
+    args = [],
+    timeout = 20000,
+    includeStderr,
+  } = opts;
+
+  if (!_.isString(command)) {
     log.errorAndThrow(`The 'command' argument is mandatory'`);
   }
-  let args = opts.args;
-  if (util.hasValue(args)) {
-    if (!_.isArray(args)) {
-      args = [args];
+
+  const adbArgs = [
+    ...this.adb.executable.defaultArgs,
+    'shell',
+    command,
+    ...(_.isArray(args) ? args : [args])
+  ];
+  log.debug(`Running '${this.adb.executable.path} ${quote(adbArgs)}'`);
+  try {
+    const {stdout, stderr} = await exec(this.adb.executable.path, adbArgs, {timeout});
+    if (includeStderr) {
```

Comment:
what's the harm in including `stderr` in the object even if it's not requested?

Reply:
I want to keep the backward compatibility